code/modules/urist/items/clothstuff.dm

```diff
@@ -94,3 +94,46 @@
 				else
 					to_chat(user, "<span class='warning'>\The [src] is used up, but there are more wounds to treat on \the [affecting.name].</span>")
 			use(used)
+
+/obj/item/loom
+	name = "table loom"
+	desc = "A loom small enough to only take up a table instead of the whole floor."
+	icon = 'icons/urist/items/wood.dmi'
+	icon_state = "loom"
+
+/datum/reagent/cottonfiber
+	name = "cotton fiber"
+	description = "A mass of cotton fibers."
+	taste_description = "cotton"
+	reagent_state = SOLID
+	color = "#f0f0f0"
+
+/datum/seed/cotton
+	name = "cotton"
+	seed_name = "cotton"
+	display_name = "cotton plant"
+	chems = list(/datum/reagent/cottonfiber = list(6,1))
+
+/datum/seed/cotton/New()
+	..()
+	set_trait(TRAIT_MATURATION,10)
+	set_trait(TRAIT_PRODUCTION,1)
+	set_trait(TRAIT_YIELD,8)
+	set_trait(TRAIT_POTENCY,10)
+	set_trait(TRAIT_PRODUCT_ICON,"cotton")
+	set_trait(TRAIT_PRODUCT_COLOUR,"#f0f0f0")
+	set_trait(TRAIT_PLANT_ICON,"bush2")
+	set_trait(TRAIT_WATER_CONSUMPTION, 6)
+
+/obj/item/seeds/cotton
+	seed_type = "cotton"
+
+/obj/item/reagent_containers/food/snacks/grown/attackby(obj/item/W, mob/user)
```

Comment:
i have to agree with Blanc on this, this seems like a very hacky and difficult to maintain way of doing this. It also just seems a bit unintuitive. Why are we calling the attackby() on the cotton itself and not the loom. Seems more intuitive to hit the loom with the cotton and not the other way around.

Reply:
fixed with your code
ty <3